custom-sort-string.java

```diff
@@ -0,0 +1,35 @@
+// Time Complexity : O(M + N)
+// Space Complexity : O(M + N)
```

Comment:
Space complexity will be constant as map will consist of constant number of characters